write-fonts/src/graph.rs

```diff
@@ -615,6 +681,63 @@ impl Graph {
         true
     }
 
+    /// for each space that has overflows and > 1 roots, select half the roots
+    /// and move them to a separate subgraph.
+    //
+    /// return `true` if any change was made.
+    ///
+    /// This is a port of the [_try_isolating_subgraphs] method in hb-repacker.
+    ///
+    /// [_try_isolating_subgraphs]: https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-repacker.hh#L182
+    fn try_splitting_subgraphs(&mut self, overflows: &[Overflow]) -> bool {
+        let mut to_isolate = HashMap::new();
+        for overflow in overflows {
+            let child_space = self.nodes[&overflow.child].space;
+            // we only isolate subgraphs in wide-space
+            if !child_space.is_custom() || self.num_roots_per_space[&child_space] <= 1 {
+                continue;
+            }
+            let root = self.find_root_of_space(overflow.child);
+            debug_assert_eq!(self.nodes[&root].space, child_space);
+            to_isolate
+                .entry(child_space)
+                .or_insert_with(HashSet::new)
+                .insert(root);
+        }
+
+        if to_isolate.is_empty() {
+            return false;
+        }
```

Comment:
At a glance it looks like you could  write this as to_isolate = overflows.iter.filter.map.collect?